cmake/board_revision_ifdef.cmake

```diff
@@ -0,0 +1,20 @@
+# SPDX-License-Identifier: Apache-2.0
```

Comment:
@TTDRosen - this is a reusable cmake rule for the `#ifdef` version of `BOARD_REVISION`